navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -0,0 +1,260 @@
+
+  private MapView mapView;
+  private InstructionView instructionView;
+
+  private MapboxMap map;
+  private MapboxNavigation navigation;
+  private NavigationMapRoute mapRoute;
+  private NavigationCamera camera;
+  private LocationEngine locationEngine;
+  private LocationLayerPlugin locationLayer;
+
+  private Location location;
+  private Position destination;
+  private boolean unpackBundle;
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.navigation_view_layout);
+    unpackBundle = savedInstanceState == null;
+    bind();
+
+    initMap(savedInstanceState);
+    initNavigation();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+    if (locationLayer != null) {
+      locationLayer.onStop();
+    }
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+    if (locationLayer != null) {
+      locationLayer.onStop();
+    }
+  }
+
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+    if (navigation != null) {
+      navigation.onDestroy();
+    }
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    map = mapboxMap;
+    initLocation();
+    initLocationLayer();
+    initRoute();
+    initCamera();
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  @Override
+  public void onConnected() {
+    locationEngine.requestLocationUpdates();
+  }
+
+  @Override
+  public void onLocationChanged(Location location) {
+    this.location = location;
+    if (unpackBundle) {
+      unpackBundle(getIntent());
+      unpackBundle = false;
+    }
+  }
+
+  @Override
+  public void onProgressChange(Location location, RouteProgress routeProgress) {
+    locationLayer.forceLocationUpdate(location);
+    instructionView.update(routeProgress);
+  }
+
+  @Override
+  public void userOffRoute(Location location) {
+    Position newOrigin = Position.fromLngLat(location.getLongitude(), location.getLatitude());
+    fetchRoute(newOrigin, destination);
+  }
+
+  @Override
+  public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
+    if (validRouteResponse(response)) {
+      startNavigation(response.body().getRoutes().get(0));
+    }
+  }
+
+  @Override
+  public void onFailure(Call<DirectionsResponse> call, Throwable t) {
+    // TODO Send alertview message
```

Comment:
Add alertview functionality or ticket this out seperately and remove the TODO

Reply:
Removed and ticketed here https://github.com/mapbox/mapbox-navigation-android/issues/203